lib/plsql/procedure_call.rb

```diff
@@ -236,6 +236,14 @@ def add_argument(argument, value, argument_metadata=nil)
         @bind_values[argument] = value.nil? ? nil : (value ? 1 : 0)
         @bind_metadata[argument] = argument_metadata.merge(:data_type => "NUMBER", :data_precision => 1)
         "l_#{argument}"
+      when 'UNDEFINED'
+        if argument_metadata[:type_name] == 'XMLTYPE'
+          @declare_sql << "l_#{argument} XMLTYPE;\n"
+          @assignment_sql << "l_#{argument} := XMLTYPE(:#{argument});\n" if not value.nil?
+          @bind_values[argument] = value if not value.nil?
+          @bind_metadata[argument] = argument_metadata.merge(:data_type => "CLOB")
+          "l_#{argument}"
+        end
```

Comment:
There are some other other `UNDEFINED` types (I'm checking on that on http://sqlfiddle.com/):
```
select
  data_type, type_name, type_subname, pls_type, count(*) as cnt
from
  all_arguments a
where
  a.data_type = 'UNDEFINED'
group by
  data_type, type_name, type_subname, pls_type
```

It's an edge case indeed, yet I'm wondering about the behavior (how gracefully does it fail) when facing some of these unsupported types.